exonum/src/proto/mod.rs

```diff
@@ -335,6 +337,77 @@ where
     }
 }
 
+impl<K, V> ProtobufConvert for exonum_merkledb::MapProof<K, V>
+where
+    K: BinaryKey,
+    V: BinaryValue,
+    Vec<(K, Option<V>)>: FromIterator<(<K as ToOwned>::Owned, Option<V>)>,
+{
+    type ProtoStruct = MapProof;
+
+    fn to_pb(&self) -> Self::ProtoStruct {
+        let mut map_proof = MapProof::new();
+
+        let proof: Vec<MapProofEntry> = self
+            .proof_unchecked()
+            .iter()
+            .map(|(p, h)| {
+                let mut entry = MapProofEntry::new();
+                entry.set_hash(h.to_pb());
+                entry.set_proof_path(p.as_bytes().to_vec());
```

Comment:
I'd suggest to extract `ProofPath` serialization into a separate place (e.g., by implementing `ProtobufConvert` for this type), and to use the same "compressed" serialization as for hashing:

```proto
message ProofPath {
  uint32 bit_count = 1;
  bytes bits = 2; // minimum necessary length, padded with zeros if necessary
}
```